src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/SimpleAttributeFeatureChainWfsTest.java

```diff
@@ -385,6 +424,169 @@ public void testClientPropertiesFilter() {
+        NestedFilterToSQL nestedFilterToSQL = createNestedFilterEncoder(rootMapping);
+
+        FilterFactoryImplNamespaceAware ff = new FilterFactoryImplNamespaceAware();
+        ff.setNamepaceContext(rootMapping.getNamespaces());
+
+        /*
+         * test filter by name coming from child table
+         */
+        PropertyIsEqualTo propertyIsEqualTo = ff.equals(ff.property("gml:name[2]"),
+                ff.literal("nameone 4"));
+
+        // Filter involving single nested attribute --> can be encoded
+        ComplexFilterSplitter splitter = new ComplexFilterSplitter(store.getFilterCapabilities(),
+                rootMapping);
+        splitter.visit(propertyIsEqualTo, null);
+        Filter preFilter = splitter.getFilterPre();
+        Filter postFilter = splitter.getFilterPost();
+
+        assertEquals(propertyIsEqualTo, preFilter);
+        assertEquals(Filter.INCLUDE, postFilter);
+
+        // filter must be "unrolled" (i.e. reverse mapped) first
+        Filter unrolled = AppSchemaDataAccess.unrollFilter(propertyIsEqualTo, rootMapping);
+
+        // Filter is nested
+        assertTrue(NestedFilterToSQL.isNestedFilter(unrolled));
+
+        String encodedFilter = nestedFilterToSQL.encodeToString(unrolled);
+
+        // this is the generated query in PostGIS, but the test limits to check the presence of the
+        // EXISTS keyword, as the actual SQL is dependent on the underlying database
+//        String expectedEncoded = "EXISTS (SELECT \"chain_link_1\".\"PKEY\" "
+//                + "FROM \"appschematest\".\"MAPPEDFEATURENAMEONE\" \"chain_link_1\" "
+//                + "WHERE \"chain_link_1\".\"NAME\" = 'nameone 4' "
+//                + "AND \"appschematest\".\"MAPPEDFEATUREWITHNESTEDNAME\".\"ID\" = \"chain_link_1\".\"MF_ID\")";
+//        assertEquals(expectedEncoded, encodedFilter);
+        assertTrue(encodedFilter.contains("EXISTS"));
+        assertContainsFeatures(fs.getFeatures(propertyIsEqualTo), "mf3");
+
+        /*
+         * test filter on attribute of root feature type
+         */
+        PropertyIsEqualTo ordinaryFilter = ff.equals(ff.property("gml:name[1]"),
+                ff.literal("GUNTHORPE FORMATION"));
+
+        // Filter involving direct attribute of root feature type --> can be encoded
+        ComplexFilterSplitter splitter2 = new ComplexFilterSplitter(store.getFilterCapabilities(),
+                rootMapping);
+        splitter2.visit(ordinaryFilter, null);
+        preFilter = splitter2.getFilterPre();
+        postFilter = splitter2.getFilterPost();
+
+        assertEquals(ordinaryFilter, preFilter);
+        assertEquals(Filter.INCLUDE, postFilter);
+
+        // Filter must be "unrolled" (i.e. reverse mapped) prior to being encoded
+        unrolled = AppSchemaDataAccess.unrollFilter(ordinaryFilter, rootMapping);
+
+        // Filter is NOT nested
+        assertFalse(NestedFilterToSQL.isNestedFilter(unrolled));
+        // IS THIS A BUG?!?
```

Comment:
@bencaradocdavies  Well, the comment is kind of a leftover, after my changes to `XPathUtil`. Taking the mapping configuration used by `SimpleAttributeFeatureChainWfsTest` as an example, before my (supposed) fix, a simple filter like:
```
[ gml:name[1] = GUNTHORPE FORMATION ]
```
would be unrolled to:
```
[[ LEX_D = GUNTHORPE FORMATION ] OR [ gml:name[1] = GUNTHORPE FORMATION ] OR [ gml:name[1] = GUNTHORPE FORMATION ] OR [ gml:name[1] = GUNTHORPE FORMATION ]]
```
The first ORed filter is correct as it is the actual "unrolling" of the `gml:name[1]` mapping, the others come from the mappings of `gml:name[2]`, `gml:name[3]` and `gml:name[4]`, which are considered a match because `XPathUtil` (used to) compare xpaths *always* ignoring indexes.

I believe this is (was) a bug, but I'd be glad to be confirmed, or corrected,  in my assessment :wink: